fix in multiple packages:  Change font size and line height to rem

HeleneKassandra opened the pull request:
## Beskrivelse
Denne fiksen setter en default font-size på html elementet, og endrer alle andre font-size og line-height verdier i designsystemet til rem. Det skal ikke være noen visuelle endringer. 

## Motivasjon og kontekst
Noen brukere hadde problemer med å forstørre teksten på iOS.
Vi ønsker at brukere skal selv kunne forstørre innholdet så mye de selv ønsker.

## Testing
Kjørte npm start, og sammenlignet localhost versjonen (med endringene) med live versjonen design.sparebank1.no
Så ingen visuelle endringer på komponentene.

HeleneKassandra committed:
fix(ffe-account-selector-react): change text values to rem

BREAKING CHANGE: This fix changes all font-size and line-height values from px to rem,
this is to solve an issue with zooming in iOS.

HeleneKassandra committed:
fix(ffe-cards): change text values to rem

BREAKING CHANGE: This fix changes all font-size and line-height values from px to rem,
this is to solve an issue with zooming in iOS.

HeleneKassandra committed:
fix(ffe-core): change text values to rem

BREAKING CHANGE: This fix changes all font-size and line-height values from px to rem,
this is to solve an issue with zooming in iOS.

HeleneKassandra committed:
fix(ffe-datepicker): change text values to rem

BREAKING CHANGE: This fix changes all font-size and line-height values from px to rem,
this is to solve an issue with zooming in iOS.

HeleneKassandra committed:
fix(ffe-file-upload): change text values to rem

BREAKING CHANGE: This fix changes all font-size and line-height values from px to rem,
this is to solve an issue with zooming in iOS.

HeleneKassandra committed:
fix(ffe-header): change text values to rem

BREAKING CHANGE: This fix changes all font-size and line-height values from px to rem,
this is to solve an issue with zooming in iOS.

HeleneKassandra committed:
fix(ffe-system-message): change text values to rem

BREAKING CHANGE: This fix changes all font-size and line-height values from px to rem,
this is to solve an issue with zooming in iOS.

HeleneKassandra committed:
fix(ffe-tables): change text values to rem

BREAKING CHANGE: This fix changes all font-size and line-height values from px to rem,
this is to solve an issue with zooming in iOS.

HeleneKassandra committed:
fix(ffe-tabs): change text values to rem

BREAKING CHANGE: This fix changes all font-size and line-height values from px to rem,
this is to solve an issue with zooming in iOS.

HeleneKassandra committed:
docs: change text values to rem

This changes font-size and line-height values from px to rem,
this was part of a larger fix which was done across the whole designsystem.
The files in this commit are only used in the documentation and not the
designsystem itself

kwltrs commented on packages/ffe-core/less/font-sizes.less:
```diff
@@ -1,3 +1,7 @@
+/* Font-size Default */
+html {
+    font-size: 16px;
+}
```
Denne regeln burde ikke være her fordi `font-sizes.less` er liksom en bibliotekfil som inneholder kun [less-mixins](http://lesscss.org/features/#mixins-feature), dvs. om man importerer denne filen i en annen less-fil så får man ingen css-regler men kan bruke de mixins `.ffe-fontsize-*()` i sin kode.
Derfor er det bedre å ha den font-size- regelen i `ffe-normalize.less`.

HeleneKassandra replied:
Jeg har ikke noe ffe-normalize.less fil?  Jeg ser den er importert, men jeg finner ikke selve filen?  Jeg finner normalize.css, men det er jo tredjepart

antidecaf replied:
Den ligger i ffe-core, her: `designsystem/packages/ffe-core/less/ffe-normalize.less`

kwltrs replied:
Har du https://github.com/SpareBank1/designsystem/blob/develop/packages/ffe-core/less/ffe-normalize.less ?

HeleneKassandra replied:
Fant den,  jeg må ha vært blind. fikset nå

kwltrs reviewed (changes requested):
Du må ha med `BREAKING CHANGE` i commitene slik at det blir en ny major-release på det pakkene. [Se eksemplet her på wiki'n](https://github.com/SpareBank1/designsystem/wiki/Bidra:-Fra-issue-til-pull-request#commit-meldinger).

HeleneKassandra commented:
> Du må ha med `BREAKING CHANGE` i commitene slik at det blir en ny major-release på det pakkene. [Se eksemplet her på wiki'n](https://github.com/SpareBank1/designsystem/wiki/Bidra:-Fra-issue-til-pull-request#commit-meldinger).

Breaking change i commit tittelen eller funker det i content? 
Og hvordan kan jeg legge til det nå i etterkant?

kwltrs commented:
> Breaking change i commit tittelen eller funker det i content?

I content.

> Og hvordan kan jeg legge til det nå i etterkant?

Jeg bruker `git rebase --interactive develop` på cli for å reworde commits. Finnes helt sikkert noe med bedre UI, men jeg likker jo terminalen ;-)

HeleneKassandra committed:
fix(ffe-core): move html font-size

BREAKING CHANGE: Moved the html font-size defintion from font-sizes.less to ffe-normalize.

HeleneKassandra commented:
> > Breaking change i commit tittelen eller funker det i content?
> 
> I content.
> 
> > Og hvordan kan jeg legge til det nå i etterkant?
> 
> Jeg bruker `git rebase --interactive develop` på cli for å reworde commits. Finnes helt sikkert noe med bedre UI, men jeg likker jo terminalen ;-)

Skal jeg ta det mot develop branchen?  Eller den branchen jeg lagde for dettte?  Jeg har dårlig erfaring med rebase

kwltrs commented:
> Skal jeg ta det mot develop branchen? Eller den branchen jeg lagde for dettte?

På din branch. Istedenfor `develop` i kommandot `git rebase --interactive develop` kan du bruke hashen til commiten i develop din branch tok utgangspunkt i. 

> Jeg har dårlig erfaring med rebase

Det har vi alle. Men også mange bra minnen ;-)

fredrikp commented:
Jeg lastet ned CSS-filen som @HeleneKassandra la ut i Slack-kanalen. Så laget jeg en html-fil med noen av overskriftene. Testet i Safari på to iPhoner, en med standard skriftstørrelse og en med XXXL skriftstørrelse. Teksten ble seende likt ut på begge telefonene, så det ser ikke ut til at denne PRen muliggjør støtte for Dynamic Type på iOS. 

Se denne artikkelen for en kjapp intro til hvordan Dynamic Type fungerer:
https://useyourloaf.com/blog/using-dynamic-type-with-web-views/

Her er Apples oversikt over predefinerte skriftstørrelser, se avsnitt "Dynamic Type Sizes":
https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/typography/

![Native oversikt](https://user-images.githubusercontent.com/888442/99533431-2fe85f80-29a6-11eb-9f54-98f9ad66cc6c.png)

fredrikp commented:
Et eksempel på hvordan Dynamic Type kan kombineres med en custom font (eksemplet er for en statisk webside+css bakt inn i en iOS-app):
```
@font-face {
    font-family: 'MuseoSans-500';
    src: local('MuseoSans-500'),url('MuseoSans_500.otf') format('opentype');
}

body {
    margin: 16pt;
    font: -apple-system-body;
    font-family: "MuseoSans-500";
}

h1 {
    font: -apple-system-title1;
}
```
Nå vil body og h1 vises med Museo Sans 500, men størrelsen vil være 17pt og 28pt for en bruker med standard skriftsstørrelse, og de vil skalere for andre skriftsstørrelser.

ivarni commented:
Dette kan fikses mye lettere med `commit` kommandoen, bare å kjøre `git commit --amend`. Rebase er overkill :)

Hm nei, det er 11 commits ja. Lat som jeg ikke sa noe... 😨

ghost commented:
Denne PR-en vil først og fremst legge til rette for at vi kan spesifisere relative fontstørrelser i kombinasjon med Dynamic Type. For å faktisk "koble seg på" Dynamic Type så kan man legge på følgende i appen som konsumerer komponenter fra designsystemet:

```css
@supports (font: -apple-system-body) {
  :root {
    font: -apple-system-body;
  }
}
```
(Merk: dette legges på `:root`, siden base font-size må ligge på en "child" av elementet som bruker `-apple-system-*` regelen, i dette tilfellet `html` som spesifisert i PR-en).

Har verifisert dette ved å laste ned/ta i bruk CSS-fila med endringane, samt legge til regelen over i appen det gjelder. Fontstørrelsen tilpasser seg da til det som er spesifisert i iOS systeminnstillingene. 

**Et mulig alternativ for å slippe å måtte gjere dette manuelt i kvar enkelt app, er å legge regelen over inn i designsystemet,
dog det er kanskje greiest å la dette vere en "opt-in" (dvs. å faktisk benytte seg av Dynamic Type blir opp til kvar einskild app å bestemme)?**

fredrikp commented:
@sondrelarsen Jeg la inn snutten med "supports" i tillegg, og da fungerer Dynamic Type i native app med Webview også. Veldig bra!

![Native oversikt](https://user-images.githubusercontent.com/888442/99639023-64602800-2a47-11eb-8334-49e05c6a7861.png)

antidecaf commented:
> **Et mulig alternativ for å slippe å måtte gjere dette manuelt i kvar enkelt app, er å legge regelen over inn i designsystemet, dog det er kanskje greiest å la dette vere en "opt-in" (dvs. å faktisk benytte seg av Dynamic Type blir opp til kvar einskild app å bestemme)?**

Høres fornuftig ut. Kanskje legge det i en egen `dynamic-type.less` som er optional?
